core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -235,4 +415,116 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        List<Object> terms = new ArrayList<>();
+        if (isTermsLookupQuery()) {
+            handleTermsLookup(termsLookup.toTermsLookup(context), context.client(), terms);
+        } else {
+            terms.addAll(values);
+        }
+        if (terms.isEmpty()) {
+            return Queries.newMatchNoDocsQuery();
+        }
+        return handleTermsQuery(terms, fieldName, context, minimumShouldMatch, disableCoord);
+    }
+
+    static void handleTermsLookup(TermsLookup lookup, Client client, List<Object> terms) {
```

Comment:
thanks for the help, i think i can manage :) Just doing step by step, first the terms and then the lookup.